drivers/sensor/bq2415x/bq2415x.c

```diff
@@ -0,0 +1,870 @@
+}
+
+/* enable/disable auto resetting chip timer */
+static int bq2415x_set_autotimer(struct bq2415x_device *bq2415x, int state)
+{
+	if (state) {
+		k_delayed_work_submit(&bq2415x->timer_reset_work, K_SECONDS(BQ2415X_TIMER_TIMEOUT));
+		int ret;
+
+		ret = bq2415x_exec_command(bq2415x, BQ2415X_TIMER_RESET);
+		if (ret < 0) {
+			LOG_ERR("Failed to reset timer");
+			return ret;
+		}
+	} else {
+		k_delayed_work_cancel(&bq2415x->timer_reset_work);
+	}
+
+	return 0;
+}
+
+/* delayed work function for auto resetting chip timer */
+static void bq2415x_timer_work(struct k_work *work)
+{
+	struct bq2415x_device *bq2415x =
+		CONTAINER_OF(work, struct bq2415x_device, timer_reset_work);
+	int ret;
+
+	ret = bq2415x_exec_command(bq2415x, BQ2415X_TIMER_RESET);
+	if (ret < 0) {
+		LOG_ERR("Failed to reset timer.");
+		bq2415x_set_autotimer(bq2415x, 0);
+		return;
+	}
+	k_delayed_work_submit(&bq2415x->timer_reset_work, K_SECONDS(BQ2415X_TIMER_TIMEOUT));
+}
+
+
+/* reset all chip registers to default state */
+static int bq2415x_reset_chip(struct bq2415x_device *bq2415x)
+{
+	int ret =
+		bq2415x_i2c_write(bq2415x, BQ2415X_REG_CURRENT, BQ2415X_RESET_CURRENT);
+
+	ret += bq2415x_i2c_write(bq2415x, BQ2415X_REG_VOLTAGE, BQ2415X_RESET_VOLTAGE);
+	ret += bq2415x_i2c_write(bq2415x, BQ2415X_REG_CONTROL, BQ2415X_RESET_CONTROL);
+	ret += bq2415x_i2c_write(bq2415x, BQ2415X_REG_STATUS, BQ2415X_RESET_STATUS);
+	return ret;
+}
+
+/**** properties functions ****/
+
+/* set current limit in mA */
+static int bq2415x_set_current_limit(struct bq2415x_device *bq2415x, int mA)
+{
+	int val;
+
+	if (mA <= 100) {
+		val = 0;
+	} else if (mA <= 500) {
+		val = 1;
+	} else if (mA <= 800) {
+		val = 2;
+	} else {
+		val = 3;
+	}
+
+	return bq2415x_i2c_write_mask(bq2415x, BQ2415X_REG_CONTROL, val,
+				      BQ2415X_MASK_LIMIT, BQ2415X_SHIFT_LIMIT);
+}
+
+/* get current limit in mA */
+static int bq2415x_get_current_limit(struct bq2415x_device *bq)
+{
+	int ret;
+
+	ret = bq2415x_i2c_read_mask(bq, BQ2415X_REG_CONTROL, BQ2415X_MASK_LIMIT,
+				    BQ2415X_SHIFT_LIMIT);
+	if (ret < 0) {
+		return ret;
+	} else if (ret == 0) {
+		return 100;
+	} else if (ret == 1) {
+		return 500;
+	} else if (ret == 2) {
+		return 800;
+	} else if (ret == 3) {
+		return 1800;
+	}
+	return -EINVAL;
+}
+
+/* set weak battery voltage in mV */
+static int bq2415x_set_weak_battery_voltage(struct bq2415x_device *bq, int mV)
+{
+	int val;
+
+	/* round to 100mV */
+	if (mV <= 3400 + 50) {
+		val = 0;
+	} else if (mV <= 3500 + 50) {
+		val = 1;
+	} else if (mV <= 3600 + 50) {
+		val = 2;
+	} else {
+		val = 3;
+	}
+
+	return bq2415x_i2c_write_mask(bq, BQ2415X_REG_CONTROL, val,
+				      BQ2415X_MASK_VLOWV, BQ2415X_SHIFT_VLOWV);
+}
+
+/* get weak battery voltage in mV */
+static int bq2415x_get_weak_battery_voltage(struct bq2415x_device *bq)
+{
+	int ret;
+
+	ret = bq2415x_i2c_read_mask(bq, BQ2415X_REG_CONTROL, BQ2415X_MASK_VLOWV,
+				    BQ2415X_SHIFT_VLOWV);
+	if (ret < 0) {
+		return ret;
+	}
```

Comment:
Add empty line after if {} block, it is inconsistent in many places, please apply it everywhere

Reply:
Done. I hope I didn't miss any.